packages/nimble-components/src/text-area/index.ts

```diff
@@ -26,6 +28,69 @@ export class TextArea extends FoundationTextArea {
      */
     @attr
     public appearance: TextAreaAppearance = TextAreaAppearance.outline;
+
+    /**
+     * A message explaining why the value is invalid.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: error-text
+     */
+    @attr({ attribute: 'error-text' })
+    public errorText?: string;
+
+    /**
+     * Whether to display the error state.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: error-visible
+     */
+    @attr({ attribute: 'error-visible', mode: 'boolean' })
+    public errorVisible = false;
+
+    /**
+     * A reference to the internal error icon element
+     * @internal
+     */
+    public errorIcon!: IconExclamationMark;
+
+    private resizeObserver?: ResizeObserver;
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.resizeObserver = new ResizeObserver(this.onResize);
+        this.resizeObserver.observe(this);
+    }
+
+    /**
+     * @internal
+     */
+    public override disconnectedCallback(): void {
+        this.resizeObserver?.disconnect();
+    }
+
+    /**
+     * @internal
+     */
+    public onTextAreaInput(): void {
+        this.handleTextInput();
+        this.updateScrollbarWidth();
+    }
+
+    private readonly onResize = (): void => {
+        this.updateScrollbarWidth();
+    };
+
+    private updateScrollbarWidth(): void {
+        this.errorIcon.style.setProperty(
+            '--ni-private-scrollbar-width',
+            `${this.control.offsetWidth - this.control.clientWidth}px`
+        );
```

Comment:
Instead of mutating the style attribute programmatically we should set this via a template binding. ie.:

```ts
this.scrollbarWidth = this.control.offsetWidth - this.control.clientWidth;
```

where `scrollbarWidth` is `@internal` and then in the template bind to the value:
```ts
`<div class="container" style="--ni-private-scrollbar-width:${x => x.scrollbarWidth}px;">`
```

And notice as described below this style is demonstrating placing it on a child of the host element, i.e. the `.container` element.

Reply:
Can see how we are aligning on the pattern with the table as-well: https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/template.ts#L22

Reply:
Changed to using a template binding as suggested.